sdccc/src/test/java/com/draeger/medical/sdccc/tests/util/MdibHistorianTest.java

```diff
@@ -905,6 +915,271 @@ void testUniqueEpisodicReportBasedHistoryFailInvalidateTestRunWhenSameReportType
+     *       in this case.
+     */
+    @Test
+    void testApplyReportOnStorageRegressionCalledWithSystemErrorReport() {
+
+        // given
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+        final var report = new SystemErrorReport();
+        final SystemErrorReport.ReportPart part = new SystemErrorReport.ReportPart();
+        part.setErrorCode(createCodedValue("someCode", "someCodingSystem", "1.0"));
+        final LocalizedText text =
+                createLocalizedText("someErrorText", "en", "someRef", BigInteger.ZERO, LocalizedTextWidth.M);
+        part.setErrorInfo(text);
+        report.getReportPart().add(part);
+        report.setMdibVersion(numericMdibVersion);
+        report.setSequenceId(sequenceId);
+
+        testApplyReportOnStorage(numericMdibVersion, sequenceId, report);
+    }
+
+    /**
+     * Tests if uniqueEpisodicReportBasedHistoryUntilTimestamp() gracefully ignores OperationInvokedReports.
+     * NOTE: this is a regression test as uniqueEpisodicReportBasedHistoryUntilTimestamp() failed in this case
+     *       before.
+     * @throws JAXBException - when thrown by the MessageStorage
+     * @throws IOException - when thrown by the MessageStorage
+     */
+    @Test
+    void testUniqueEpisodicReportBasedHistoryUntilTimestampRegressionWithOperationInvokedReport()
+            throws JAXBException, IOException {
+
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+
+        final var report = messageBuilder.buildOperationInvokedReport(sequenceId);
+        final var part = messageBuilder.buildOperationInvokedReportReportPart();
+        final var invInfo = messageBuilder.buildInvocationInfo(123, InvocationState.FAIL);
+        invInfo.setInvocationError(InvocationError.OTH);
+        part.setInvocationInfo(invInfo);
+        part.setOperationTarget("opTarget");
+        part.setOperationHandleRef("opHandle");
+
+        final var invSrc = mdibBuilder.buildInstanceIdentifier();
+        final var type = mdibBuilder.buildCodedValue("33");
+        type.setCodingSystem("JOCS");
+        type.setCodingSystemVersion("2.0");
+        invSrc.setType(type);
+        invSrc.setRootName("rootName");
+        invSrc.setExtensionName("extName");
+        part.setInvocationSource(invSrc);
+        part.setSourceMds("sourceMds");
+        report.getReportPart().add(part);
+        report.setMdibVersion(numericMdibVersion);
+        final Envelope soapMessage =
+                messageBuilder.createSoapMessageWithBody(ActionConstants.ACTION_OPERATION_INVOKED_REPORT, report);
+
+        testUniqueEpisodicReportBasedHistoryUntilTimestamp(sequenceId, soapMessage);
+    }
+
+    /**
+     * Tests if uniqueEpisodicReportBasedHistoryUntilTimestamp() gracefully ignores SystemErrorReports.
+     * NOTE: this is a regression test as uniqueEpisodicReportBasedHistoryUntilTimestamp() failed in this case
```

Comment:
I've looked into this and that is not true, same for `testEpisodicReportBasedHistoryRegressionWithSystemErrorReport()`. Both mentioned methods from the MdibHistorian (`uniqueEpisodicReportBasedHistoryUntilTimestamp()` and `episodicReportBasedHistory()`) are filtering the reports by type. removing SystemErrorReports (see `Constants.RELEVANT_REPORT_BODIES`).

Reply:
True, this is the comment for testUniqueEpisodicReportBasedHistoryUntilTimestampRegressionWithOperationInvokedReport, that was copied together with the method and turned into testUniqueEpisodicReportBasedHistoryUntilTimestampRegressionWithSystemErrorReport.
For OperationInvokedReport it is true, but you are right, for SystemErrorReport it is not.
I removed this note for the two methods.